### PR TITLE
48 merchants revenue

### LINKED ISSUE #48: GET /api/v1/merchants/revenue?date=x

### PR DESCRIPTION
**Pull request by ski-climb**
closes #48 

Whew!  The setup on these things is killer.  A good next step would be to extract the setup for the three tests in the merchant revenue spec file.  The test is on the verbose side, but I was struggling to find a more succinct way to test all the various edge cases (i.e. different date, unsuccessful transaction, invoice_item not on an invoice that we were including...).

@kheppenstall ready for review

### CODE REVIEW

File: spec/models/invoice_spec.rb [hunk 1]
```diff
@@ -23,4 +23,10 @@
       expect(invoice.status).to eq "shipped"
     end
   end
+
+  describe ".revenue_by_day" do
+    it "responds to revenue_by_day"do
+      expect(Invoice).to respond_to(:revenue_by_day)
+    end
+  end
 end
```
Comment (kheppenstall) [inline spec/models/invoice_spec.rb:32]: I like this test. Let's make sure we add these in for all of our other business intelligence methods.

File: spec/requests/api/v1/business-intelligence/merchant_revenue_spec.rb [hunk 2]
```diff
@@ -45,10 +43,37 @@
...
+
+    it 'returns the total revenue' do
+      expected_revenue = 66_66 + 42_24
+      get "/api/v1/merchants/revenue?date=#{sliver_of_time}"
+
+      expect(response).to be_success
+      expect(Invoice.revenue_by_day(sliver_of_time)).to eq expected_revenue
+    end
+  end
 end
```
Comment (kheppenstall) [inline spec/requests/api/v1/business-intelligence/merchant_revenue_spec.rb:79]: Nice work testing unsuccessful and successful transactions.

File: app/models/invoice.rb [hunk 3]
```diff
@@ -11,9 +11,16 @@ class Invoice < ApplicationRecord
   def self.on_date(date)
     if date
       day = Date.parse(date)
-      where(invoices: {created_at: day.beginning_of_day..day.end_of_day})
+      where(invoices: {created_at: date})
```
Comment (saltybigmac) [inline app/models/invoice.rb:14]: you are within the `Invoice` so this just needs to be where(created_at: date)

File: app/models/invoice.rb [hunk 4]
```diff
@@ -11,9 +11,16 @@ class Invoice < ApplicationRecord
...
       all
     end
   end
+
+  def self.revenue_by_day(day)
+      joins(:transactions, :invoice_items)
+      .merge(Transaction.successful)
+      .merge(self.on_date(day))
+      .sum('unit_price_in_cents * quantity')
+  end
```
Comment (saltybigmac) [inline app/models/invoice.rb:25]: This method should really live within the `Merchant`. Also, you are merging self but that's not needed. Move this into the `Merchant` and you should be able to clean this up a bit.
